x/merkledb/db.go

```diff
@@ -655,15 +660,47 @@ func (db *merkleDB) GetChangeProof(
 	return result, nil
 }
 
-// NewView returns a new view on top of this trie.
+// NewViewFromBatchOps returns a new view on top of this trie.
+// Changes made to the view will only be reflected in the original trie if Commit is called.
+// if copyBytes is true, code will duplicate any passed []byte so that editing in calling code is safe
+// Assumes [db.commitLock] and [db.lock] aren't held.
+func (db *merkleDB) NewViewFromBatchOps(_ context.Context, batchOps []database.BatchOp, copyBytes bool) (TrieView, error) {
+	return db.newView(func() (*trieView, error) {
+		return newTrieViewFromBatchOps(db, db, db.root.clone(), batchOps, copyBytes)
+	})
+}
+
+// Returns a new view that isn't tracked in [db.childViews].
+// For internal use only, namely in methods that create short-lived views.
+// Assumes [db.lock] isn't held and [db.commitLock] is read locked.
+func (db *merkleDB) newUntrackedView(batchOps []database.BatchOp, copyBytes bool) (*trieView, error) {
+	return newTrieViewFromBatchOps(db, db, db.root.clone(), batchOps, copyBytes)
+}
+
+// NewViewFromMap returns a new view on top of this trie.
 // Changes made to the view will only be reflected in the original trie if Commit is called.
+// if copyBytes is true, code will duplicate any passed []byte so that editing in calling code is safe
 // Assumes [db.commitLock] and [db.lock] aren't held.
-func (db *merkleDB) NewView(_ context.Context, batchOps []database.BatchOp) (TrieView, error) {
+func (db *merkleDB) NewViewFromMap(_ context.Context, changes map[string]*ChangeOp, copyBytes bool) (TrieView, error) {
```

Comment:
I don't think we need a new type here
```suggestion
func (db *merkleDB) NewViewFromMap(_ context.Context, changes map[string]maybe.Maybe[[]byte], copyBytes bool) (TrieView, error) {
```